app/controllers/application_controller.rb

```diff
@@ -120,6 +121,24 @@ def service_unavailable(exception = nil)
     true
   end
 
+  def sti_clean_up(e)
+    @unknown_class_name = e.message.match(/subclass: '(.*)'\./)[1]
+    @parent_class = e.message.match(/overwrite (.*)\.inheritance_column/)[1].constantize
+    if params[:confirm_data_deletion] == 'yes'
+      begin
+        @parent_class.where(@parent_class.inheritance_column => @unknown_class_name).delete_all
```

Comment:
Shouldn't this be possible only for admins?

Reply:
I'm sure I planned to limit that but I probably forgot at the end, of course this entire page should only be available to admin, normal users should see regular 500